mempool/src/core_mempool/index.rs

```diff
@@ -3,27 +3,126 @@
+pub struct AccountTransactions {
+    nonce_transactions: BTreeMap<u64 /* Nonce */, MempoolTransaction>,
+    sequence_number_transactions: BTreeMap<u64 /* Sequence number */, MempoolTransaction>,
+}
+
+impl AccountTransactions {
+    pub(crate) fn get(&self, replay_protector: &ReplayProtector) -> Option<&MempoolTransaction> {
+        match replay_protector {
+            ReplayProtector::Nonce(nonce) => self.nonce_transactions.get(nonce),
+            ReplayProtector::SequenceNumber(sequence_number) => {
+                self.sequence_number_transactions.get(sequence_number)
+            },
+        }
+    }
+
+    pub(crate) fn get_mut(
+        &mut self,
+        replay_protector: &ReplayProtector,
+    ) -> Option<&mut MempoolTransaction> {
+        match replay_protector {
+            ReplayProtector::Nonce(nonce) => self.nonce_transactions.get_mut(nonce),
+            ReplayProtector::SequenceNumber(sequence_number) => {
+                self.sequence_number_transactions.get_mut(sequence_number)
+            },
+        }
+    }
+
+    pub(crate) fn insert(&mut self, txn: MempoolTransaction) {
+        match txn.sequence_info.transaction_replay_protector {
+            ReplayProtector::Nonce(nonce) => {
+                self.nonce_transactions.insert(nonce, txn);
+            },
+            ReplayProtector::SequenceNumber(sequence_number) => {
+                self.sequence_number_transactions
+                    .insert(sequence_number, txn);
+            },
+        }
+    }
+
+    pub(crate) fn remove(
+        &mut self,
+        replay_protector: &ReplayProtector,
+    ) -> Option<MempoolTransaction> {
+        match replay_protector {
+            ReplayProtector::Nonce(nonce) => self.nonce_transactions.remove(nonce),
+            ReplayProtector::SequenceNumber(sequence_number) => {
+                self.sequence_number_transactions.remove(sequence_number)
+            },
+        }
+    }
+
+    pub(crate) fn append(&mut self, other: &mut Self) {
+        self.nonce_transactions
+            .append(&mut other.nonce_transactions);
+        self.sequence_number_transactions
+            .append(&mut other.sequence_number_transactions);
+    }
+
+    pub(crate) fn clear(&mut self) {
+        self.nonce_transactions.clear();
+        self.sequence_number_transactions.clear();
+    }
+
+    pub(crate) fn seq_num_split_off(&mut self, sequence_number: u64) -> Self {
+        AccountTransactions {
+            sequence_number_transactions: self
+                .sequence_number_transactions
+                .split_off(&sequence_number),
+            nonce_transactions: mem::take(&mut self.nonce_transactions),
+        }
+    }
+
+    pub(crate) fn seq_num_range_mut(
+        &mut self,
+        range: impl RangeBounds<u64>,
+    ) -> RangeMut<'_, u64, MempoolTransaction> {
+        self.sequence_number_transactions.range_mut(range)
+    }
+
+    pub(crate) fn values(&self) -> impl Iterator<Item = &MempoolTransaction> {
```

Comment:
this implicitly prioritizes orderless txn over sequence number txns per account, not sure if it's intended

Reply:
Yeah, orderless transactions can be pulled from mempool into QS without any extra waiting time (like parking lot). So, in the mempool priority index, I gave higher preference for orderless transaction when compared to sequence number based transaction if they have same ranking score (gas price).